gplugins/lumerical/fdtd.py

```diff
@@ -0,0 +1,1282 @@
+"""Write Sparameters with Lumerical FDTD"""
+from __future__ import annotations
+
+import multiprocessing
+import shutil
+import time
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import gdsfactory as gf
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import yaml
+from gdsfactory.component import Component
+from gdsfactory.config import __version__, logger
+from gdsfactory.pdk import get_layer_stack
+from gdsfactory.technology import LayerStack
+
+from gplugins.common.utils.get_sparameters_path import (
+    get_sparameters_path_lumerical as get_sparameters_path,
+)
+from gplugins.lumerical.config import marker_list, um
+from gplugins.lumerical.convergence_settings import (
+    LUMERICAL_FDTD_CONVERGENCE_SETTINGS,
+    ConvergenceSettingsLumericalFdtd,
+)
+from gplugins.lumerical.simulation_settings import (
+    SIMULATION_SETTINGS_LUMERICAL_FDTD,
+    SimulationSettingsLumericalFdtd,
+)
+from gplugins.lumerical.utils import draw_geometry, layerstack_to_lbr
+
+try:
+    import lumapi
+except ModuleNotFoundError as e:
+    print(
+        "Cannot import lumapi (Python Lumerical API). "
+        "You can add set the PYTHONPATH variable or add it with `sys.path.append()`"
+    )
+    raise e
+except OSError as e:
+    raise e
+
+if TYPE_CHECKING:
+    from gdsfactory.typings import PathType
+
+run_false_warning = """
+You have passed run=False to debug the simulation
+
+run=False returns the simulation session for you to debug and make sure it is correct
+
+To compute the Sparameters you need to pass run=True
+"""
+
+
+def main():
```

Comment:
examples should go outside of the library file

Reply:
Moved to examples:
4fe395fa77f5e5f571b9f308771cabcc2138fb9c